distributed/shuffle/_core.py

```diff
@@ -104,12 +104,22 @@ async def barrier(self) -> None:
     async def send(
         self, address: str, shards: list[tuple[_T_partition_id, bytes]]
     ) -> None:
-        self.raise_if_closed()
-        return await self.rpc(address).shuffle_receive(
-            data=to_serialize(shards),
-            shuffle_id=self.id,
-            run_id=self.run_id,
-        )
+        retries_left = 3
```

Comment:
There's also `distributed.utils_comm.retry` which includes some features like an exponential back-off. If you haven't already done this, I'd suggest taking a look at it and see if we can leverage it here.

Reply:
> For reference, would we expect a user to be setting `distributed.comm.retry.delay.min|max` or `distributed.comm.retry.count` in some way that wouldn't play well with P2P?

I double-checked `distributed.comm.retry.count`, which defaults to `0`. This would make using `retry_operation` rather pointless unless users explicitly set this.

I'm fine with setting sensible defaults for now or adding something like `distributed.p2p.retry.*` to the configuration. I leave this up to you.